src/main/webapp/js/restaurant-page-loader.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Get ?user=XYZ parameter value
+const urlParams = new URLSearchParams(window.location.search);
+const parameterUsername = urlParams.get('restaurant');
+
+// URL must include ?user=XYZ parameter. If not, redirect to homepage.
+if (!parameterUsername) {
+  window.location.replace('/');
+}
+
+/** Sets the page title based on the URL parameter username. */
+function setPageTitle() {
+  document.getElementById('page-title').innerText = parameterUsername;
+  document.title = parameterUsername + ' - User Page';
+}
+
+/**
+ * Shows the message form if the user is logged in and viewing their own page.
+ */
+function showMessageFormIfViewingSelf() {
+  fetch('/login-status')
+      .then((response) => {
+        return response.json();
+      })
+      .then((loginStatus) => {
+        if (loginStatus.isLoggedIn &&
+            loginStatus.username == parameterUsername) {
+          const messageForm = document.getElementById('message-form');
+          messageForm.classList.remove('hidden');
+          document.getElementById('about-me-form').classList.remove('hidden');
+        }
+      });
+}
+
+/** Fetches about me messages and add them to the page*/
+function fetchAboutMe(){
+  const url = '/about?user=' + parameterUsername;
+  fetch(url).then((response) => {
+    return response.text();
+  }).then((aboutMe) => {
+    const aboutMeContainer = document.getElementById('about-me-container');
+    if(aboutMe == ''){
+      aboutMe = 'This user has not entered any information yet.';
+    }
+
+    aboutMeContainer.innerHTML = aboutMe;
+
+  });
+}
+
+/** Fetches messages and add them to the page. */
+function fetchMessages() {
+  const url = '/messages?user=' + parameterUsername;
```

Comment:
Should this and the above method be fetching for a user ID instead of a restaurant ID?